src/sage/combinat/sf/character.py

```diff
@@ -36,7 +36,8 @@
 from sage.rings.integer import Integer
 
 
-class generic_character(SFA_generic):
+class Character_generic(SFA_generic):
```

Comment:
Is deprecation needed for this renaming?

Reply:
Not really, because `generic_character` was never intended as a "public" class.  The symmetric function basis would always be obtained using `SymmetricFunctions(QQ).ht` (or `.st`).

Reply:
OK, fine with me.